.github/workflows/build.yml

```diff
@@ -2,7 +2,6 @@ name: Build
 
 on:
   push:
-    branches: [main]
   pull_request:
```

Comment:
Didn't you realize that you don't need to keep the pull_request part when all branches are covered elsewhere?